src/Aquifer.Jobs/Subscribers/TranslationSubscriber.cs

```diff
@@ -0,0 +1,379 @@
+            .Where(rcv => dto.TranslatedProjectResourceContentIds.Contains(rcv.ResourceContentId)))
+        {
+            // now that ALL translations are complete for the project, assign all resources in the project to the company lead
+            resourceContentVersion.AssignedUserId = project.CompanyLeadUserId;
+            await _resourceHistoryService.AddAssignedUserHistoryAsync(
+                resourceContentVersion,
+                assignedUserId: project.CompanyLeadUserId,
+                changedByUserId: dto.StartedByUserId,
+                activityContext.CancellationToken);
+        }
+
+        await _dbContext.SaveChangesAsync(activityContext.CancellationToken);
+
+        await _notificationService.SendProjectStartedNotificationAsync(dto.ProjectId, activityContext.CancellationToken);
+    }
+
+    /// <summary>
+    /// Note: Callers are responsible for performing resource assignments and creating original resource content snapshots.
+    /// This method will perform the actual translation, create machine translations,
+    /// update the existing draft resource version with translated content, create a new snapshot for the translation,
+    /// and change the resource content status.
+    /// </summary>
+    private async Task TranslateResourceCoreAsync(
+        int resourceContentId,
+        int startedByUserId,
+        TranslationOrigin translationOrigin,
+        CancellationToken ct)
+    {
+        _logger.LogInformation(
+            "Beginning translation for Resource Content ID {ResourceContentId} (started by User ID {StartedByUserId} via Translation Origin: {TranslationOrigin}).",
+            resourceContentId,
+            startedByUserId,
+            translationOrigin);
+
+        var resourceContentVersion = await _dbContext.ResourceContentVersions
+            .AsTracking()
+            .Include(rcv => rcv.ResourceContent)
+            .Include(rcv => rcv.ResourceContent.Language)
+            .Include(rcv => rcv.ResourceContentVersionSnapshots)
+            .SingleOrDefaultAsync(
+                rcv =>
+                    rcv.IsDraft &&
+                    rcv.ResourceContent.MediaType == ResourceContentMediaType.Text &&
+                    rcv.ResourceContent.Id == resourceContentId,
+                ct)
+            ?? throw new InvalidOperationException(
+                $"Aborting translation for Resource Content ID {resourceContentId} because it has no Resource Content Version with Text Media Type in the Draft status.");
+
+        if (resourceContentVersion.ResourceContentVersionSnapshots.Count == 0)
+        {
+            throw new InvalidOperationException(
+                $"Aborting translation for Resource Content ID {resourceContentId} because there are no pre-existing Snapshots for Resource Content Version ID {resourceContentVersion.Id}.");
+        }
+
+        if (resourceContentVersion.ResourceContent.Status != ResourceContentStatus.TranslationAwaitingAiDraft)
```

Comment:
The exception above can only happen if we're in an invalid state because anything that queues a translation message should have already created a snapshot for all related resource content versions.

Being that such a state should never happen, it seems right to validate the data before attempting any action on it, including gracefully skipping translation.